fastlane/lanes/screenshots.rb

```diff
@@ -192,4 +211,33 @@
       download_path: options[:download_path]
     )
   end
+
+  # Create simulators needed for the `screenshots` lane if they don't exist yet
+  #
+  # Note: Be sure to call this lane after `xcodebuild` has been invoked at least once in the VM / local machine
+  # because on a fresh VM where Xcode has never launched yet, there might be no simulator at all, but as soon as
+  # Xcode is launched for the first time it will create a set of defaut simulators.
+  #
+  # We only want to create the missing simulators for screenshots if they don't exist in the default set created
+  # after the first `xcodebuild` launch, otherwise if we create a simulator while it will also be created by Xcode
+  # later, we'll end up with `The requested device could not be found because multiple devices matched the request.`
+  # when we try to run `capture_ios_screenshots`
```

Comment:
It would be nice to find away to automate a guard against this occurrence, but I think it's safe to leave here because it relates to specialized tooling that 1) few people work with and 2) is automated, therefore easier to run in a way that accounts for this note 👍

Reply:
Another idea I thought about was to create the simulators in all cases, but then (1) either reference them with their UUID when passing it to the `devices:`, or alternatively (2) create them with distinctive names like `"StoreScreenshots - iPhone 8 Plus"` (and passing those to `devices:`

But I wasn't sure if that would risk impacting the name of the generated files (raw PNG screenshots) created by fastlane during each run of the UITests, so preferred to keep it at that for the first iteration.

(I might try experimenting with it again later depending on how the rest of the project goes… 🤞 )